js/src/checkout-drop-in.js

```diff
@@ -1,246 +1,23 @@
-	 */
-	const validate_response = response => {
-		return response.json()
-			.then( data => {
-				if ( 200 !== response.status ) {
-					throw new Error( data.message, {
-						cause: data
-					} );
-				}
-
-				return data;
-			} );
-	};
-
-	/**
-	 * Process response.
-	 *
-	 * @param data Object from JSON response.
-	 */
-	const process_response = data => {
-		// Handle action object.
-		if ( data.action ) {
-			dropin.handleAction( data.action );
-
-			return;
-		}
-
-		// Handle result code.
-		if ( data.resultCode ) {
-			paymentResult( data );
-		}
-	}
-
-	/**
-	 * Handle error.
-	 *
-	 * @param error
-	 */
-	const handle_error = error => {
-		// Check syntax error name.
-		if ( 'SyntaxError' === error.name ) {
-			error.message = pronamicPayAdyenCheckout.syntaxError;
-		}
-
-		// Show error message.
-		dropin.setStatus( 'error', { message: error.message } );
-	}
-
-	/**
-	 * Get payment methods configuration.
-	 *
-	 * @return object
-	 */
-	const getPaymentMethodsConfiguration = () => {
-		/**
-		 * Complement Apple Pay configuration with `onValidateMerchant`
-		 * callback when using your own Apple Pay certificate.
-		 * 
-		 * @link https://github.com/pronamic/wp-pronamic-pay-adyen/issues/5#issue-1154083692
-		 */
-		if ( pronamicPayAdyenCheckout.paymentMethodsConfiguration.applepay && pronamicPayAdyenCheckout.applePayMerchantValidationUrl ) {
-			pronamicPayAdyenCheckout.paymentMethodsConfiguration.applepay.onValidateMerchant = ( resolve, reject, validationUrl ) => {
-				send_request(
-					pronamicPayAdyenCheckout.applePayMerchantValidationUrl,
-					{
-						validation_url: validationUrl
-					}
-				)
-				.then( validate_response )
-				.then( data => {
-					// Handle Apple error.
-					if ( data.statusMessage ) {
-						throw new Error( data.statusMessage, {
-							cause: data
-						} );
-					}
-
-					resolve( data );
-				} )
-				.catch( error => {
-					handle_error( error );
-
-					// Reject to dismiss Apple Pay overlay.
-					reject( error );
-				} );
-			};
-		}
-
-		// Compliment PayPal configuration.
-		if ( pronamicPayAdyenCheckout.paymentMethodsConfiguration.paypal ) {
-			pronamicPayAdyenCheckout.paymentMethodsConfiguration.paypal.onCancel = ( data, dropin ) => {
-				dropin.setStatus( 'ready' );
-			};
-		}
+'use strict';
 
-		return pronamicPayAdyenCheckout.paymentMethodsConfiguration;
-	};
+( async function () {
 
 	/**
 	 * Adyen Checkout.
 	 */
-	const checkout = new AdyenCheckout( pronamicPayAdyenCheckout.configuration );
-
-	const dropin = checkout.create( 'dropin', {
-		paymentMethodsConfiguration: getPaymentMethodsConfiguration(),
-		onSelect: dropin => {
-			let configuration = pronamicPayAdyenCheckout.configuration;
-
-			if ( false === configuration.showPayButton ) {
-				dropin.submit();
-			}
+	const configuration = {
+		...pronamicPayAdyenCheckout.configuration,
+		onPaymentCompleted: ( result, component ) => {
+			window.location.href = pronamicPayAdyenCheckout.paymentReturnUrl;
```

Comment:
If the payment is refused, e.g. with incorrect card details, I'm being redirected to the payment return URL. However, the payment status is not updated to **Failed** and does not have the failure reason set.

For all card payments (and possibly also with other payment methods), the redirect results in the below JSON response:

```json
{"code":"rest_missing_callback_param","message":"Ontbrekende parameter(s): redirectResult","data":{"status":400,"params":["redirectResult"]}}
```

Reply:
Improved in https://github.com/pronamic/wp-pronamic-pay-adyen/commit/b7c8bd1677be3e1f1e5eb8cd8ecfe9986a58e265.